src/lib/registration/sw_mgmt.rb

```diff
@@ -126,7 +126,12 @@ def self.find_base_product
       # on a running system the products are :installed
       # during upgrade use the newer selected product (same as in installation)
       products = Pkg.ResolvableProperties("", :product, "").find_all do |p|
-        if Stage.initial
+        if Stage.initial && Mode.auto
+          Yast.import "AutoinstConfig"
+          # note: AutoinstConfig.selected_product should never be nil when
+          # AY let it pass here
+          p["name"] == AutoinstConfig.selected_product.name
```

Comment:
Is the `AutoinstConfig.selected_product.name` value always set? What if it is missing in the profile? I'm afraid of the `nil` value here...

Reply:
If no product can be recognized from the profile then AY currently reports error and aborts installation. If this behavior is changed this needs to be updated too.

Reply:
I see, so it is checked earlier. OK.